Model 2.0 filtering

frostyfan109 opened the pull request:
- Adds arbitrary property filtering for DugElement fields on search endpoints
- Adds arbitrary property aggregation for DugElement fields on search endpoints
- Updates `/concepts` endpoint to utilize `search_elements` method.
  - Removes `concept_types` request/response fields; superseded by new `aggs` param.
- Remove `_count` call in `search_elements` in favor of `hits.total`

frostyfan109 committed:
add arbitrary proprety filtering for dug elements

frostyfan109 committed:
adding aggregations support

frostyfan109 committed:
update faceted pagination

frostyfan109 committed:
add size filtering, update remove_hits_from_results to handle empty hits object

frostyfan109 committed:
add exists, missing ops; update /concepts endpoint to use search_elements and support filters/aggs

frostyfan109 committed:
update size filters to support objects

frostyfan109 committed:
remove debug trycatch

frostyfan109 committed:
update endpoint docstrings

hina-shah commented:
This is great! 

Can you add to the documentation allowable operator values for the filters. I had to go back to the code to find allowable filters.

hina-shah reviewed (approved):
Looks good. Not sure about the need for aggregations yet but I'll let @YaphetKG  comment. 
Once the doc has been updated, we can go ahead and merge.

frostyfan109 committed:
update doc comments about available operators and clarify subfield usage

frostyfan109 commented:
> Can you add to the documentation allowable operator values for the filters. I had to go back to the code to find allowable filters.

I updated the docs with a nested bullet of available operators for the `filters` parameter. I also added a note for `filters` and `aggs` about using subfields like `.keyword` when referencing text fields.

> Looks good. Not sure about the need for aggregations yet but I'll let @YaphetKG comment. Once the doc has been updated, we can go ahead and merge.

Aggregations are important for implementing faceted search on the frontend, it should come in handy for David
